tensorflow/python/framework/graph_util_impl.py

```diff
@@ -123,6 +123,11 @@ def extract_sub_graph(graph_def, dest_nodes):
   if not isinstance(graph_def, graph_pb2.GraphDef):
     raise TypeError("graph_def must be a graph_pb2.GraphDef proto.")
 
+  # Github 13047, cast single string to list
+  if isinstance(dest_nodes, str):
+    dest_nodes = [dest_nodes]
+    logging.warning("dest_nodes should be a list.")
```

Comment:
Raise a TypeError instead. We shouldn't have a special case for passing in a string.